src/language-handlebars/printer-glimmer.js

```diff
@@ -454,6 +519,31 @@ function clean(ast, newObj) {
   }
 }
 
+function countNewLines(string) {
+  /* istanbul ignore next */
+  string = typeof string === "string" ? string : "";
+  (string.match(/[\r\n]+/g) || [])[0];
+  return string.split("\n").length - 1;
+}
+
+function countLeadingNewLines(string) {
+  /* istanbul ignore next */
+  string = typeof string === "string" ? string : "";
+  const newLines = (string.match(/^[\r\n]+/g) || [])[0] || "";
+  return countNewLines(newLines);
+}
+
+function countTrailingNewLines(string) {
+  /* istanbul ignore next */
+  string = typeof string === "string" ? string : "";
+  const newLines = (string.match(/[\r\n]+$/g) || [])[0] || "";
```

Comment:
Same as above, newlines with whitespaces should still be newlines:
`string.match(/([\r\n][^\S\r\n]*)+$/g)` should work